src/providers/walletconnectNotify.ts

```diff
@@ -0,0 +1,146 @@
+import fetch from 'node-fetch';
+import { capture } from '@snapshot-labs/snapshot-sentry';
+import { getSubscribers } from '../helpers/utils';
+
+const WALLETCONNECT_NOTIFY_SERVER_URL = process.env.WALLETCONNECT_NOTIFY_SERVER_URL;
+const WALLETCONNECT_PROJECT_SECRET = process.env.WALLETCONNECT_PROJECT_SECRET;
+const WALLETCONNECT_PROJECT_ID = process.env.WALLETCONNECT_PROJECT_ID;
+
+const AUTH_HEADER = {
+  Authorization: WALLETCONNECT_PROJECT_SECRET ? `Bearer ${WALLETCONNECT_PROJECT_SECRET}` : ''
+};
+
+// Match Snapshot event names to notification types
+// That should be defined in the wc-notify-config.json
+function getNotificationType(event) {
+  if (event.includes('proposal/')) {
+    return 'ed2fd071-65e1-440d-95c5-7d58884eae43';
+  } else {
+    return null;
+  }
+}
+
+function getNotificationBody(event, space) {
+  switch(event) {
+  case "proposal/create":
+    return `A new proposal has been created for ${space.name}`
+  case "proposal/end":
+    return `A proposal has closed for ${space.name}`
+  default:
+    return null;
+  }
+}
+
+// Fetch subscribers from WalletConnect Notify server
+export async function getSubscribersFromWalletConnect() {
+  const fetchSubscribersUrl = `${WALLETCONNECT_NOTIFY_SERVER_URL}/${WALLETCONNECT_PROJECT_ID}/subscribers`;
+
+  try {
+    const subscribersRs = await fetch(fetchSubscribersUrl, {
+      headers: AUTH_HEADER
+    });
+
+    const subscribers: string[] = await subscribersRs.json();
+
+    return subscribers;
+  } catch (e) {
+    capture('[WalletConnect] failed to fetch subscribers');
+    return [];
+  }
```

Comment:
Why are we catching and suppressing this error? If we can't get subscribers then the functions that call this should fail as well, not proceed with an empty list.

Reply:
the other providers like discord and xmtp don't throw  and just end up being a noop so I did the same here